db/flush_job.cc

```diff
@@ -454,18 +448,12 @@ Status FlushJob::MemPurge(autovector<MemTable*>& purged_mems) {
         break;
       }
 
-      // If new_mem is full, add it to purged_mems, and allocate new
-      // memtable.
-      if (new_mem->ShouldScheduleFlush()) {
-        // Rectify the first sequence number, which (unlike the earliest seq
-        // number) needs to be present in the new memtable.
-        new_mem->SetFirstSequenceNumber(new_first_seqno);
-        purged_mems.push_back(new_mem);
-        new_mem =
-            new MemTable((cfd_->internal_comparator()), *(cfd_->ioptions()),
-                         mutable_cf_options_, cfd_->write_buffer_mgr(),
-                         mems_[0]->GetEarliestSequenceNumber(), cfd_->GetID());
-        new_first_seqno = kMaxSequenceNumber;
+      // If new_mem has size greater than maxSize,
+      // then rollback to regular flush operation,
+      // and destroy new_mem.
+      if (new_mem->ApproximateMemoryUsage() > maxSize) {
+        s = Status::Aborted(Slice("Mempurge filled more than one memtable."));
```

Comment:
I don't think you need `Slice(` due to implicit conversions char[] -> char* -> std::string -> Slice

Reply:
Fixed